python/cudf_polars/cudf_polars/experimental/scheduler.py

```diff
@@ -0,0 +1,148 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+"""Synchronous task scheduler."""
+
+from __future__ import annotations
+
+from collections.abc import MutableMapping
+from typing import TYPE_CHECKING, Any, TypeVar
+
+if TYPE_CHECKING:
+    from collections.abc import Mapping
+    from typing import TypeAlias
+
+
+Key: TypeAlias = str | tuple[str, int]
+Graph: TypeAlias = MutableMapping[Key, Any]
+T_ = TypeVar("T_")
+
+
+# NOTE: This is a slimmed-down version of the single-threaded
+# (synchronous) scheduler in `dask.core`.
+#
+# Key Differences:
+# * We do not allow a task to contain a list of key names.
+#   Keys must be distinct elements of the task.
+
+
+def istask(x: Any) -> bool:
+    """Check if x is a callable task."""
+    return bool(type(x) is tuple and x and callable(x[0]))
```

Comment:
```suggestion
    return type(x) is tuple and x and callable(x[0])
```

Reply:
mypy doesn't like the `... and x ...`, but we can just do `... and bool(x) ...`